core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -373,7 +375,15 @@ private void handleRequest(ChannelHandlerContext ctx, DecodedHttpRequest req) th
             serviceResponse = serviceResponse.recover(cause -> {
                 reqCtx.logBuilder().responseCause(cause);
                 // Recover the failed response with the error handler.
-                return serviceCfg.errorHandler().onServiceException(reqCtx, cause);
+                final HttpResponse recovered = serviceCfg.errorHandler().onServiceException(reqCtx, cause);
+                if (recovered == null) {
+                    return null;
+                }
+                final ResponseContentPreviewer previewer = responseContentPreviewer(reqCtx);
+                if (previewer == null) {
+                    return recovered;
+                }
+                return previewer.setUp(recovered);
```

Comment:
No confident that this is the right way to fix the issue.
I had a second idea which introduces a hook `mapOnRecover` and set up the response content previewer on the hook using `peek`:
```java
final HttpResponse res = unwrap().serve(ctx, req);
final HttpResponse previewRes = res.peek(setResponseContentPreviewer());
previewRes.mapOnRecover(recovered -> {
    return recovered.peek(setUpResponseContentPreviewer(...));
});
return previewRes;
```
We might also use `mapOnRecover` applying the same encoding to the recovered response when an `EncodingService` is decorated:
```java
final HttpResponse encodedRes = new HttpEncodedResponse(...);
return encodedRes.mapOnRecover(recovered -> new HttpEncodedResponse(...));
```
But I was not sure that the hook is a desirable public API because it seemed to me that the hook would cause the recovery behavior to run all over the place, and make it hard to predict the recovery result in `HttpServerHandler`.
So before making lots of changes without confidence, sent this PR with the smallest fixes.

Reply:
Not sure what is a good solution for this. I'll give it some thought. ⌛️

Reply:
I think it's pretty useful to have `mapOnRecover` just like the example @ghkim3221 shows. cc also @trustin, @jrhee17

Reply:
@ikhoon Could you share the idea that adds the previewer to the `ServerBuilder` and `ClientBuilder` to @ghkim3221?